### PR TITLE
Add interpreter for SqrtOp.

### LINKED ISSUE #1116: Add interpreter for SqrtOp
Following the [testing guidelines](https://github.com/openxla/stablehlo/blob/main/docs/interpreter_checklist.md)

### PR DESCRIPTION
**Pull request by subhankarshah**
closes #1116 

We have the following constraints in the spec:

```
(I1) operand is a tensor of floating-point or complex type.
(C1) operand and result have the same type.
```

These constraints are covered by the following tests

```
(I1) operand is not a tensor of floating-point or complex type. (covered by ODS)
(C1) operand and result do not have the same type. (covered by ODS)
```

### CODE REVIEW

File: stablehlo/tests/interpret_square_root.mlir [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  // CHECK-NEXT: 5.0328918128646478 : f64
+  // CHECK-NEXT: 0xFFF8000000000000 : f64
+  // CHECK-NEXT: 1.000000e+00 : f64
+  // CHECK-NEXT: 0.000000e+00 : f64
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: square_root_op_test_c64
+func.func @square_root_op_test_c64() -> tensor<2xcomplex<f32>> {
```
Comment (ghpvnist) [inline stablehlo/tests/interpret_square_root.mlir:18]: According to the [testing_guidelines](https://github.com/openxla/stablehlo/blob/main/docs/reference.md#testing-guidelines), we should be testing with the maximum bit-width for each type:
```
func.func @square_root_op_test_c128() -> tensor<2xcomplex<f64>> {
```

**Review by ghpvnist — CHANGES_REQUESTED** [review]
I encourage you to refer to our [reference_checklist](https://github.com/openxla/stablehlo/blob/main/docs/reference_checklist.md) to ensure there's no missing files.

**Review by ghpvnist — CHANGES_REQUESTED** [review]
I made similar comment to #1008. PTAL

**Review by ghpvnist — APPROVED** [review]

**Review by sdasgup3 — APPROVED** [review]
For completeness add the testing strategy (https://github.com/openxla/stablehlo/blob/main/docs/reference_checklist.md). In this case, we will end up with no up if we follow  https://github.com/openxla/stablehlo/pull/996#issue-1558631158

**Comment by ghpvnist** [comment]
Linking issue: [Make sure that examples in the spec/ODS conform to spec_checklist.md](#454)

File: stablehlo/dialect/StablehloOps.td [hunk 2]
```diff
@@ -604,9 +604,9 @@ def StableHLO_SineOp: StableHLO_UnaryElementwiseOp<"sine",
   }];
 }
 
-def StableHLO_SqrtOp: StableHLO_UnaryElementwiseOp<"sqrt",
-    [Pure, HLO_CompatibleOperandsAndResultType], HLO_FpOrComplexTensor> {
-  let summary = "Square-root operator";
+def StableHLO_SqrtOp: StableHLO_UnaryElementwiseOp<"sqrt", [Pure,
+    HLO_CompatibleOperandsAndResultType /*sqrt_c1*/], HLO_FpOrComplexTensor> {
+  let summary = "Sqrt operation";
```
Comment (sdasgup3) [inline stablehlo/dialect/StablehloOps.td:609]: Add input constraints as per  the checklist.

File: docs/spec.md [hunk 3]
```diff
@@ -5091,12 +5091,10 @@ Performs element-wise square root operation on `operand` tensor and produces a
 // %operand: [[0.0, 1.0], [4.0, 9.0]]
```
Comment (burmako) [inline docs/spec.md:5091]: Please update the PR description to follow the testing guidelines, e.g. similarly to #1080.

**Review by burmako — CHANGES_REQUESTED** [review]

**Review by burmako — APPROVED** [review]

**Review by sdasgup3 — APPROVED** [review]

**Pull request merged**